department-of-veterans-affairs/caseflow
# [APPEALS-37094] Updating regional_office_timezone method

Resolves [APPEALS-37094](https://jira.devops.va.gov/browse/APPEALS-37094)

# Description
Create a [Hearing#regional_office_timezone](https://github.com/department-of-veterans-affairs/caseflow/blob/084cd9819ca5e2151cd6daadb0273075d362cf66/app/models/hearing.rb) method where it returns the regional office timezone as per the current values returned by the regional_office_timezone method

## Acceptance Criteria
- If the hearing has a non-nil scheduled_in_timezone value:
   - The scheduled_in_timezone value should be provided as the parameter to Time.use_zone so that the hearing's date and time will be calculated using the correct reference point.
- If the hearing has a nil scheduled_in_timezone value:
   - The time zone for use as a parameter to Time.use_zone should be calculated as it stands today utilizing the RO time zone of the last user to update a hearing.
- The return value should be as follows:
   - If the associated hearing_day is nil:
     - Return nil
   - Otherwise:
      - Return an [ActiveSupport::TimeWithZone](https://api.rubyonrails.org/v5.2/classes/ActiveSupport/TimeWithZone.html) object.
   - Sample: Fri, 15 Dec 2023 14:30:00 EST -05:00

## Testing Plan
1. Go to [APPEALS-39382](https://jira.devops.va.gov/browse/APPEALS-39382) or list them below

# Backend
# Best practices
## Code Documentation Updates

## Tests

1. In Caseflow, try to Schedule a Veteran's hearing by selecting the "Schedule Veteran" button
2. Try to create a hearing for a location not based in EST
3. After selecting the appellant, select the "Schedule Veteran" option in the Actions dropdown. You will be directed to the "Schedule Appellant for a Hearing" page
4. On the "Schedule Apellant for a Hearing" page, right-click to inspect the element on the Hearing Time dropdown component. Select an option from the dropdown and the value should show the entire time string including the timezone context, similar to what is displayed in the screenshot.
5. Fill out the rest of the Schedule Veteran form and click the "Schedule" button
6. Go to the Daily Docket page for the hearing you've just scheduled and verify that the time being displayed is correct for the timezone.

### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist
### Logging

### Error Handling

### Exception Handling
